src/csharp/GeneratorParams.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntimeGenAI
+{
+    public class GeneratorParams : IDisposable
```

Comment:
We need to follow IDisposable pattern here according to the MS Docs.
Introduce bool _disposed = false; member.
Check in the beggining of the Dispose() and return.
Do not check for the handle to be Zero.

This is a simple class, but generally, Dispose() takes care of 2 classes of resources.
1) disposable members, those are taken care of when disposing argument is true. In this case, we do not have such yet.
2) Native resources. They are disposed unconditionally, but once because _diposed guards the entrance.

Set _disposed to true.

The reason for the above is a complex dynamic and interactions between GC, finalizer thread and the runtime.
See here https://learn.microsoft.com/en-us/dotnet/api/system.idisposable?view=net-8.0
and the link at the bottom how to implement IDisposable.